src/main/java/edu/greenblitz/robotName/subsystems/Arm/EndEffector/Roller/SimulationRoller.java

```diff
@@ -0,0 +1,34 @@
+package edu.greenblitz.robotName.subsystems.Arm.EndEffector.Roller;
+
+import edu.greenblitz.robotName.RobotConstants;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.wpilibj.simulation.FlywheelSim;
+
+public class SimulationRoller implements IRoller{
+	
+	private FlywheelSim motorSimulation;
+	private double appliedOutput;
+	
+	public SimulationRoller() {
+		motorSimulation = new FlywheelSim(
+				DCMotor.getNEO(RollerConstants.SimulationConstants.NUMBER_OF_MOTORS),
+				RollerConstants.SimulationConstants.GEAR_RATIO,
+				RollerConstants.SimulationConstants.MOMENT_OF_INERTIA
+		);
+	}
+	
+	@Override
+	public void setPower(double power) {
+		double voltage = power * RobotConstants.SimulationConstants.BATTERY_VOLTAGE;
+		appliedOutput = MathUtil.clamp(voltage, -RobotConstants.SimulationConstants.MAX_MOTOR_VOLTAGE, RobotConstants.SimulationConstants.MAX_MOTOR_VOLTAGE);
+		motorSimulation.setInputVoltage(appliedOutput);
+	}
+	
+	@Override
+	public void updateInputs(RollerInputsAutoLogged rollerInputs) {
+		rollerInputs.appliedOutput = appliedOutput;
+		rollerInputs.outputCurrent = motorSimulation.getCurrentDrawAmps();
+		rollerInputs.velocity = motorSimulation.getAngularVelocityRPM();
```

Comment:
how can there e a velocity only in the simulation? consider deleting the velocity